src/cpp/src/llm_pipeline.cpp

```diff
@@ -65,6 +68,22 @@ std::pair<EncodedResults, int32_t> beam_search(
 );
 
 class StatefulLLMPipeline final : public LLMPipelineImplBase {
+private:
+    void slice_matmul_statefull_model(std::shared_ptr<ov::Model> model) {
+        auto last_node = model->output(0).get_node()->input_value(0).get_node();
+        if (auto matmul = dynamic_cast<ov::op::v0::MatMul*>(last_node)) {
+            auto shape = matmul->input(0).get_partial_shape();
+            if (shape.rank().get_length() == 3 && shape[1] != 1) {
```

Comment:
why do we need a check for shape[1] (sequence length) ? I should be -1 by default as model is exported as dynamic one

Reply:
this check will throw exception if rank or shape[1] dim are dynamic

Reply:
I"d like to check if the same transformation was already applied.
But I suppose the right option is to compare with PartialShape("0..1")